src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration.Tests/samples/Sample7_ConfiguringPipeline.cs

```diff
@@ -39,10 +39,10 @@ public async Task ConfiguringPipeline()
             );
 
             // add a policy (custom behavior) that executes once per client call
-            options.AddPerCallPolicy(new AddHeaderPolicy());
+            options.BeforePolicies.Add(new AddHeaderPolicy());
 
             // add a policy that executes once per retry
-            options.AddPerRetryPolicy(new CustomLogPolicy());
+            options.AfterPolicies.Add(new CustomLogPolicy());
```

Comment:
The names `BeforePolicies` and `AfterPolicies` don't seem to offer much context.  From this sample, I don't understand what their purpose is nor when I'd add a policy using one over the other.

Reply:
I'm not a fan either, it's terrible name. BeforePolicies are the once we prepend to the pipeline, AfterPolicies the ones we append to the end. Any ideas on better names?

Reply:
Maybe `options.PrependPolicy` and `options.AppendPolicy`?    or  adopt the list semantics for `Insert`  with something like `options.InsertPolicy(policy)` and `options.InsertPolicy(position, policy)`?

Reply:
Or, maybe just the straightforward `options.AddFirstPolicy()` and `options.AddLastPolicy()` ... but I'm not a huge fan of those, either.